tests/pool_test.go

```diff
@@ -623,14 +622,16 @@ func newPoolFromVMI(vmi *v1.VirtualMachineInstance) *poolv1.VirtualMachinePool {
 	return pool
 }
 
-func notDeletedVMs(poolName string, vms *v1.VirtualMachineList) (notDeleted []v1.VirtualMachine) {
-	nonDeletedVms := tests.NotDeletedVMs(vms)
-	for _, vm := range nonDeletedVms {
-		for _, ref := range vm.OwnerReferences {
-			if ref.Name == poolName {
-				notDeleted = append(notDeleted, vm)
+func filterPoolNotDeletedVMs(poolName string, vms *v1.VirtualMachineList) []v1.VirtualMachine {
+	var notDeleted []v1.VirtualMachine
+	for _, vm := range vms.Items {
+		if vm.DeletionTimestamp == nil {
+			for _, ref := range vm.OwnerReferences {
```

Comment:
@orelmisan I integrated here the old `NotDeletedVMs()`, I don't see a reason why to loop twice, but let me know if I'm missing something.